### PR TITLE
Added correct xyz offsets to VOI creation, added xy offsets to empty cube creation [build]

### PR DESCRIPTION
**Pull request by Hewwho**

### CODE REVIEW

File: pytrip/cube.py [hunk 1]
```diff
@@ -217,18 +221,13 @@ def check_compatibility(a, b):
...
-        if a.dimx != b.dimx:
-            return False
-        elif a.dimy != b.dimy:
-            return False
-        elif a.dimz != b.dimz:
-            return False
-        elif (a.pixel_size - b.pixel_size) > eps:
-            return False
-        elif a.slice_distance != b.slice_distance:
+        if a.dimx != b.dimx or \
```
Comment (grzanka) [inline pytrip/cube.py:224]: Replace with new variables:

```
x_dim_compatible = (a.dimx == b.dimx)
y_dim_compatible = (a.dimy == b.dimy)
(...)
return x_dim_compatible and y_dim_compatible
```

File: pytrip/cube.py [hunk 2]
```diff
@@ -253,19 +252,20 @@ def slice_to_z(self, slice_number):
...
-
-        """
-        # TODO why eq not being used ?
-        # eq = util.evaluator(equation)
-        # TODO why data not being used ?
-        # data = np.array(np.zeros((self.dimz, self.dimy, self.dimx)))
-        x = np.linspace(0.5, self.dimx - 0.5, self.dimx) * self.pixel_size - center[0]
-        y = np.linspace(self.dimx - 0.5, 0.5, self.dimx) * self.pixel_size - center[1]
-        xv, yv = np.meshgrid(x, y)
+    # TODO unfinished function
```
Comment (grzanka) [inline pytrip/cube.py:255]: Remove completely and create an issue

File: pytrip/cube.py [hunk 3]
```diff
@@ -495,11 +500,11 @@ def _read_trip_header_file(self, header_path):  # TODO: could be made private? #
         # load plain of gzipped file
         if header_path.endswith(".gz"):
             import gzip
-            fp = gzip.open(header_path, "rt")
+            with gzip.open(header_path, "rt") as fp:
```
Comment (grzanka) [inline pytrip/cube.py:503]: Will it work for python 2 ?

File: pytrip/cube.py [hunk 4]
```diff
@@ -782,7 +787,7 @@ def _set_z_table_from_dicom(self, dcm):
         # if yes, then all references to whether this is sorted or not can be removed hereafter
         # (see also pytripgui) /NBassler
         self.slice_pos = []
-        for i, dcm_image in enumerate(dcm["images"]):
+        for _, dcm_image in enumerate(dcm["images"]):
             self.slice_pos.append(float(dcm_image.ImagePositionPatient[2]))
```
Comment (grzanka) [inline pytrip/cube.py:791]:
```suggestion
        for dcm_image in dcm["images"]:
            self.slice_pos.append(float(dcm_image.ImagePositionPatient[2]))
```

File: pytrip/vdx.py [hunk 5]
```diff
@@ -555,6 +557,7 @@ def create_cube(cube, name, center, width, height, depth):
     return v
 
 
+# most likely doesn't work correctly
```
Comment (grzanka) [inline pytrip/vdx.py:560]: Add an issue

File: pytrip/cube.py (hunk 3 above)
Comment (Hewwho) [reply pytrip/cube.py]: Changed it to reading binary and then decoding the data, so that it works with both 2 and 3.

**Review by grzanka — APPROVED** [review]
Looks good

**Pull request merged**